### PR TITLE
V2 model

### PR DESCRIPTION
**Pull request by nickchapman-da**
Comments and feedback please!
Few things still to do - marked by TODOs in the DAML
And also I plan to write explicit update contracts to migrate the V1 model.

### CODE REVIEW

File: project/V2/daml/Davl.daml [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+
+    -- TODO: contract key so we can find the current Worker contract
+
+    -- company can change a workers boss
+    choice Worker_Change_Boss : ContractId Worker
+      with
+        newBoss : Party
+      controller company
+        do
+          create Worker with worker; boss = newBoss; company
```
Comment (shayne-fletcher-da) [inline project/V2/daml/Davl.daml]: What happens to the old worker/boss contract, existing allocations?

File: project/V2/daml/Test.daml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  boss <- getParty "someBoss"
+  worker <- getParty "minion1"
+
+  pb <- submit company $ create Propose_Boss with b = Boss with boss; company
+  b <- submit boss $ exercise pb Propose_Boss_Accept
+
+  pw <- submit company $ exercise b Boss_Propose_Worker with worker
+  w <- submit worker $ exercise pw Propose_Worker_Accept
+
+  -- Found bug in model - need to create initial Allocation for worker! - fixed
```
Comment (shayne-fletcher-da) [inline project/V2/daml/Test.daml]: 😁

File: project/V2/daml/Davl.daml (hunk 1 above)
Comment (nickchapman-da) [reply project/V2/daml/Davl.daml]: The old Worker contract is archived by this choice.
The Boss contracts are not affected.
Allocation are unaffected, and still accessible as they are keyed on `worker : Party`

File: project/V2/daml/Test.daml (hunk 2 above)
Comment (nickchapman-da) [reply project/V2/daml/Test.daml]: Yeah. I though it would be helpful to note the bugs that I found when writing the scenario.

**Pull request merged**